src/binding/petsc4py/src/petsc4py/PETSc/DMDA.pyx

```diff
@@ -400,10 +957,30 @@ cdef class DMDA(DM):
 
     #
 
-    def setRefinementFactor(self,
-                            refine_x=2,
-                            refine_y=2,
-                            refine_z=2):
+    def setRefinementFactor(
+        self,
+        refine_x: int | None = 2,
```

Comment:
Why `| None` ? It is not the first time I found this, please double-check the rest of the file